File: src/main/java/de/tomjuri/armageddon/util/KeyBindUtil.java
package de.tomjuri.armageddon.util;

import de.tomjuri.armageddon.Armageddon;
import de.tomjuri.armageddon.mixin.MinecraftInvoker;
import net.minecraft.client.settings.KeyBinding;

public class KeyBindUtil {
    public static void leftClick() { ((MinecraftInvoker) Ref.mc()).invokeClickMouse(); }
    public static void rightClick() { ((MinecraftInvoker) Ref.mc()).invokeRightClickMouse(); }
    public static void jump() {
        new Thread(() -> {
            KeyBindUtil.setKey(Ref.mc().gameSettings.keyBindJump.getKeyCode(), true);
            SleepUtil.sleep(100);
            KeyBindUtil.setKey(Ref.mc().gameSettings.keyBindJump.getKeyCode(), false);
        }).start();
    }
    public static void setKey(int key, boolean pressed) { KeyBinding.setKeyBindState(key, pressed); }
}

File: src/main/kotlin/de/tomjuri/armageddon/util/KeyBindUtil.kt
package de.tomjuri.armageddon.util

import de.tomjuri.armageddon.mixin.MinecraftInvoker
import net.minecraft.client.settings.KeyBinding

object KeyBindUtil {
    fun leftClick() = (mc as MinecraftInvoker).invokeClickMouse()
    fun middleClick() = (mc as MinecraftInvoker).invokeMiddleClickMouse()
    fun rightClick() = (mc as MinecraftInvoker).invokeRightClickMouse()
    fun jump() = Thread {
        setPressed(gameSettings.keyBindJump, true);
        Thread.sleep(100);
        setPressed(gameSettings.keyBindJump, false)
    }.start()
    fun setPressed(key: KeyBinding, pressed: Boolean) = KeyBinding.setKeyBindState(key.keyCode, pressed)
}